### PR TITLE
added functionality for admin

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DetailsOrderController extends HttpServlet {
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]:
```suggestion
public class GetOrderDetailsController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.mixshop.model.Order;
+import com.internet.mixshop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAdminAllOrdersController extends HttpServlet {
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java]:
```suggestion
public class GetAllOrdersAdminController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAdminAllProductsController extends HttpServlet {
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java]:
```suggestion
public class GetAllProductsAdminController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/view/order/admin-orders.jsp [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/view/order/admin-orders.jsp]: Should this reflect what we show on the page? For example, "Orders"? :)

**Review by AnaLanda — COMMENTED** [review]
Good job! I left a couple minor comments.

**Review by matviichuk92 — APPROVED** [review]
good job!

File: src/main/java/com/internet/mixshop/controller/CompleteOrderController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.mixshop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long cartId = Long.parseLong(req.getParameter("id"));
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/CompleteOrderController.java]: we can use USER_ID here and not pass cartId as a parameter

File: src/main/java/com/internet/mixshop/controller/CompleteOrderController.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        orderService.completeOrder(shoppingCartService.getById(cartId));
+        resp.sendRedirect("/products");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/CompleteOrderController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/products");
```
let's use this construction everywhere - it improves code flexibility

File: src/main/java/com/internet/mixshop/controller/DeleteOrderByAdminController.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.internet.mixshop.lib.Injector;
+import com.internet.mixshop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteOrderByAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/DeleteOrderByAdminController.java]:
```suggestion
public class DeleteOrderController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/DeleteProductController.java [hunk 8]
```diff
@@ -19,6 +19,6 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         String productId = req.getParameter("id");
         Long id = Long.valueOf(productId);
         productService.delete(id);
-        resp.sendRedirect(req.getContextPath() + "/products");
+        resp.sendRedirect(req.getContextPath() + "/admin/products");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/DeleteProductController.java:22]: same for AddProductController, right?

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: I agree - this looks better

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Order order = orderService.getById(Long.valueOf(id));
+        List<Product> products = order.getProducts();
+        req.setAttribute("order", order);
+        req.setAttribute("orderProducts", products);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: line 26 is redundant - we can pass only `order`

File: src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java (hunk 2 above)
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java]:
```suggestion
public class GetAllOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: done

File: src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java (hunk 3 above)
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java]:
```suggestion
public class ManageProductsController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java]: done

File: src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java]: done

File: src/main/webapp/WEB-INF/view/order/admin-orders.jsp (hunk 4 above)
Comment (mikhail771) [reply src/main/webapp/WEB-INF/view/order/admin-orders.jsp]: done

File: src/main/java/com/internet/mixshop/controller/GetCartProductsController.java [hunk 10]
```diff
@@ -22,8 +22,9 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
         List<Product> products = shoppingCart.getProducts();
-        req.setAttribute("products", products);
         double sum = shoppingCartService.findTotalPrice(shoppingCart);
+        req.setAttribute("products", products);
+        req.setAttribute("cartId", shoppingCart.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/GetCartProductsController.java]: let's better not show cartId to our users

File: src/main/webapp/WEB-INF/view/order/admin-orders.jsp (hunk 4 above)
Comment (KseniiaMakarova) [reply src/main/webapp/WEB-INF/view/order/admin-orders.jsp]: I agree, let's add informative titles

File: src/main/webapp/WEB-INF/view/order/admin-orders.jsp [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+</head>
+<body>
+<h1>All orders</h1>
+<table border="1">
+    <tr>
+        <th>ID</th>
+        <th>Action</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/view/order/admin-orders.jsp]: let's add `Details` option as well

File: src/main/webapp/WEB-INF/view/product/all-products.jsp [hunk 12]
```diff
@@ -26,12 +26,14 @@
...
                 <a href="${pageContext.request.contextPath}/cart/products/add?id=${product.id}">To cart</a>
-                <a href="${pageContext.request.contextPath}/products/delete?id=${product.id}">Delete</a>
             </td>
         </tr>
     </c:forEach>
 </table>
 <a href="${pageContext.request.contextPath}/products/add">Add product</a>
+<a href="${pageContext.request.contextPath}/orders">My orders</a>
+<a href="${pageContext.request.contextPath}/admin/orders">Admin orders</a>
+<a href="${pageContext.request.contextPath}/admin/products">Admin products</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/view/product/all-products.jsp]: let's remove this links from here
this page will be available to user - we don't want him to go on these pages

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -113,4 +113,64 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>user-orders</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.GetAllUserOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>user-orders</servlet-name>
+        <url-pattern>/orders</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -113,4 +113,64 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>admin-orders</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.GetAdminAllOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>admin-orders</servlet-name>
+        <url-pattern>/admin/orders</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -113,4 +113,64 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>admin-delete-order</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.DeleteOrderByAdminController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>admin-delete-order</servlet-name>
+        <url-pattern>/admin/order/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -113,4 +113,64 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>admin-products</servlet-name>
+        <servlet-class>com.internet.mixshop.controller.GetAdminAllProductsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>admin-products</servlet-name>
+        <url-pattern>/admin/products</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/mixshop/controller/CompleteOrderController.java (hunk 5 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/CompleteOrderController.java]: done

File: src/main/java/com/internet/mixshop/controller/CompleteOrderController.java (hunk 6 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/CompleteOrderController.java]: done

File: src/main/java/com/internet/mixshop/controller/DeleteProductController.java (hunk 8 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DeleteProductController.java:22]: done

File: src/main/webapp/WEB-INF/view/product/all-products.jsp (hunk 12 above)
Comment (mikhail771) [reply src/main/webapp/WEB-INF/view/product/all-products.jsp]: done

File: src/main/webapp/web.xml (hunk 15 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 16 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java (hunk 9 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: done

File: src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/GetAdminAllOrdersController.java]: done

File: src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/GetAdminAllProductsController.java]: done

File: src/main/webapp/WEB-INF/view/order/admin-orders.jsp (hunk 11 above)
Comment (mikhail771) [reply src/main/webapp/WEB-INF/view/order/admin-orders.jsp]: done

File: src/main/java/com/internet/mixshop/controller/GetCartProductsController.java (hunk 10 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/GetCartProductsController.java]: done

File: src/main/webapp/web.xml (hunk 13 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 14 above)
Comment (mikhail771) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/mixshop/controller/DeleteOrderByAdminController.java (hunk 7 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DeleteOrderByAdminController.java]: done

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DetailsOrderController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]:
```suggestion
public class OrderDetailsController extends HttpServlet {
```

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.mixshop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Order order = orderService.getById(Long.valueOf(id));
+        List<Product> products = order.getProducts();
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: do you need this line (24)?

File: src/main/webapp/WEB-INF/view/order/order-details.jsp [hunk 19]
```diff
@@ -0,0 +1,27 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/view/order/order-details.jsp]: let's add informative title here

File: src/main/webapp/WEB-INF/view/product/all-products.jsp [hunk 20]
```diff
@@ -26,12 +26,12 @@
             </td>
             <td>
                 <a href="${pageContext.request.contextPath}/cart/products/add?id=${product.id}">To cart</a>
-                <a href="${pageContext.request.contextPath}/products/delete?id=${product.id}">Delete</a>
             </td>
         </tr>
     </c:forEach>
 </table>
 <a href="${pageContext.request.contextPath}/products/add">Add product</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/view/product/all-products.jsp]: is user able to add new products to storage?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java (hunk 17 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: done

File: src/main/java/com/internet/mixshop/controller/DetailsOrderController.java (hunk 18 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/controller/DetailsOrderController.java]: done

File: src/main/webapp/WEB-INF/view/order/order-details.jsp (hunk 19 above)
Comment (mikhail771) [reply src/main/webapp/WEB-INF/view/order/order-details.jsp]: done

File: src/main/webapp/WEB-INF/view/product/all-products.jsp (hunk 20 above)
Comment (mikhail771) [reply src/main/webapp/WEB-INF/view/product/all-products.jsp]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**